lib/bitmask.rb

```diff
@@ -90,4 +90,29 @@ def set_array(array)
       set attr, array.include?(attr)
     end
   end
+
+  # Most Significant Bit - MSB
+  def msb
+    # To find the position of
+    # the most significant
+    # set bit
+    k = Integer(Math.log(to_i, 2))
+
+    # To return the value
+    # of the number with set
+    # bit at k-th position
+    1 << k
+  end
+
+  # Least Significant Bit - LSB
+  # 00010100    &                (n = 20)
+  # 00010011                     (n-1 = 19)
+  # ~~~~~~~~
+  # 00010000    ^                (XOR result number with n)
+  # 00010100
+  # ~~~~~~~~
+  # 00000100  ——  LSB
```

Comment:
Neat, where did you find this algorithm?

Reply:
I went through a ton of blog posts, SO posts, and etc, looking for an ideal solution.  MSB and LSB each have multiple potential solutions, with varying degress of complexity and efficiency.  I'll try to find the source of this one.